temporal/api/workflowservice/v1/request_response.proto

```
@@ -851,6 +860,9 @@ message GetSystemInfoResponse {
 
         // True if server supports upserting workflow memo
         bool upsert_memo = 7;
+
+        // True if server supports eager workflow task dispatching for the StartWorkflowExecution API
+        bool eager_workflow_start = 8;
```

If server doesn't support this feature, the `eager_workflow_task` field will simply be null? Do we still need this feature flag in that case?

Is there any design doc I can read more about this feature and also the eager start for child workflows?

---

Yes, because there's no better way for the client to know if the server support this.
We don't have a way to eagerly start child workflows but I'll provide a design doc for eager workflow start in slack.

---

> Yes, because there's no better way for the client to know if the server support this.

I think it's important to spell out why the client needs to know ahead of time. If it's just for our test cases, ok (though I disagree but don't care much).

---

I'd say this is primarily for testing, not just ours, users could leverage this flag as well.
If my app requires this functionality, it's better to fail fast at startup than to start a defunct app.